study.py

```diff
@@ -5,31 +5,34 @@
 
 
 def study():
-    sim = Simulator(0)
+    config = Configuration(0, 64, 256, 0, 100, 20, 8, 16, 4, 0, 10, 1)
+    sim = Simulator(config)
     sim.initLogger()
-    maxTries = 10
-    step = 20
     frRange = []
+    results = []
     resultRange = []
     simCnt = 0
     sim.logger.info("Starting simulations:", extra=sim.format)
     start = time.time()
-    for fr in range(0, 100, step):
-        if fr % 10 == 0:
-            sim.logger.info("Failure rate %d %% ..." % fr, extra=sim.format)
+    for fr in range(config.failureRateStart, config.failureRateStop+1, config.failureRateStep):
         sim.resetFailureRate(fr)
-        result = 0
-        for i in range(maxTries):
-            sim.initValidators()
-            sim.initNetwork()
-            result += sim.run()
-            simCnt += 1
+        for chi in range(config.chiStart, config.chiStop+1, config.chiStep):
+            sim.resetChi(chi)
+            blockAvailable = 0
+            for run in range(config.runStart, config.runStop, config.runStep):
```

Comment:
for long simulations it is better (at least in my experience) to make run the external cycle. That means you start to see approx. but complete results earlier, and get more statistical relevance as time passes.

Reply:
Done. I also added iterating over the number of validators, network degree and block size.